app/address/[address]/layout.tsx

```diff
@@ -286,10 +291,23 @@ function AccountHeader({
             const tokenMetadata = create(metadataExtension.state, TokenMetadata);
             const { metadataAddress } = create(metadataPointerExtension.state, MetadataPointer);
 
+            (async () => {
+                try {
+                    // Avoid re-renders once tokenData has been set
+                    if (!tokenData.name && !tokenData.image) {
+                        const response = await fetch(tokenMetadata.uri);
+                        setTokenData(await response.json());
```

Comment:
since we don't trust this URI, probably makes sense to unpack the response and only take the fields we want

Reply:
This layout is called in every single page load of the Explorer. If there's an error fetching the URI it will brick loading the rest of the page. 

Better way to do this is to refactor all of this in the same pattern as the `<CompressedNftAccountHeader/>` component. Once we're inside this `IIFE` block, then return a `T22AccountHeader` component.

In the `T22AccountHeader` component, instead of executing a naked fetch, try using `useMetadataJsonLink` from `@/app/providers/compressed-nft`.